kratos/solving_strategies/schemes/residual_based_bossak_velocity_scheme.h

```diff
@@ -0,0 +1,749 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:   Jordi Cotela
+//                  Suneth Warnakulasuriya
+//
+
+#if !defined(KRATOS_RESIDUAL_BASED_BOSSAK_VELOCITY_SCHEME_H_INCLUDED)
+#define KRATOS_RESIDUAL_BASED_BOSSAK_VELOCITY_SCHEME_H_INCLUDED
+
+// System includes
+#include <vector>
+
+// External includes
+
+// Project includes
+#include "includes/checks.h"
+#include "includes/define.h"
+#include "includes/kratos_parameters.h"
+#include "solving_strategies/schemes/scheme.h"
+#include "utilities/derivatives_extension.h"
+#include "utilities/indirect_scalar.h"
+
+namespace Kratos
+{
+///@name Kratos Classes
+///@{
+
+/// A scheme for dynamic adjoint equations, using Bossak time integration.
+/**
+ * It can be used for either first- or second-order time derivatives. Elements
+ * and conditions must provide a specialization of AdjointExtensions via their
+ * data value container, which allows the scheme to operate independently of
+ * the variable arrangements in the element or condition.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class ResidualBasedBossakVelocityScheme : public Scheme<TSparseSpace, TDenseSpace>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedBossakVelocityScheme);
+
+    typedef Scheme<TSparseSpace, TDenseSpace> BaseType;
+
+    typedef typename BaseType::TSystemMatrixType SystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType SystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef std::vector<Dof<double>::Pointer> DofsVectorType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+
+    ResidualBasedBossakVelocityScheme(const double AlphaBossak,
+                                      const bool UpdateAcceleration = true,
+                                      const bool UpdateDisplacement = true)
+    {
+        mAlphaBossak = AlphaBossak;
+        mUpdateAcceleration = UpdateAcceleration;
+        mUpdateDisplacement = UpdateDisplacement;
+
```

Comment:
Can you initialize mBossak in some safe/meaningful way? Leaving it like this they will contain garbage values

Reply:
done :)